Source/WebCore/loader/DocumentWriter.cpp

```diff
@@ -232,7 +232,7 @@ bool DocumentWriter::begin(const URL& urlReference, bool dispatch, Document* own
                 document->inheritPolicyContainerFrom(parentFrame->document()->policyContainer());
                 document->checkedContentSecurityPolicy()->updateSourceSelf(parentFrame->document()->securityOrigin());
             }
-        } else if (triggeringAction && triggeringAction->requester()) {
+        } else if (triggeringAction && triggeringAction->requester() && !(document->loader() && document->loader()->substituteData().isValid())) {
```

Comment:
I'd recommend documenting these two conditions via a function or lambda. That way it's a little more clear why we are skipping this branch for alternate HTML loads. 

Something like this might be more clear:

```C++
else if (triggeringAction && triggeringAction->requester() && !isLoadingAlternateHTML())
```

Reply:
`isLoadingBrowserControlledHTML` would probably be nice, actually. I'm sure there's some name that helps convey to future people what the intent is.

Reply:
That certainly does make it easier to read. Updated.